nautobot_netbox_importer/diffsync/models/abstract.py

```diff
@@ -546,6 +547,44 @@ class ComponentModel(CustomFieldModelMixin, NautobotBaseModel):
     label: str
     description: str
 
+    @root_validator
+    def invalid_type_to_other(cls, values):  # pylint: disable=no-self-argument,no-self-use
+        """
+        Default invalid `type` fields to use `other` type.
+
+        Almost all Component Models have a `type` field that uses a ChoiceSet to limit valid
+        choices. This uses Pydantic's root_validator to clean up the `type` data before loading
+        it into a Model instance. All invalid types will be changed to "other."
+
+        Two things must be done in order for a Component Model to inherit this behavoir:
+
+          1. The Model must define a `type` field.
+          2. The Model must define a class variable named `_type_choices`.
+
+        The `_type_choices` variable should be a set of the valid choices for the type per
+        Nautobot.
+        """
+        if "type" not in cls.__fields__ or "_type_choices" not in cls.__dict__:
```

Comment:
Is this the cleanest way to do this, or would something like `if not hasattr(cls, "type") or not hasattr(cls, "_type_choices")` work?

Reply:
actually, the Pydantic definitions are not class attributes, and so `hasattr(cls, "type")` will fail. However, I think we can simplify this just to check for `_type_choices` and any implementation that opts in has to define a type field (which why define _type_choices without having a type?)